### PR TITLE
Feat/plug in pyvista as plotter

### PR DESCRIPTION
**Pull request by prmukherj**
Refactor PyVisualization to plug-in plotters as required by user.

Earlier, the graphics renderer code was dependent on "PyVista" and the plotter on "matplotlib" respectively. Now, they have been separated out of the infrastructure. For example, we have introduced a pyvista plotter as an alternative to matplotlib using an abstract class providing the basic infrastructure.

Most of the exposed user-interface remains unaffected by this though.

### CODE REVIEW

**Comment by mkundu1** [comment]
Can you please add some description of the code change?

File: README.rst [hunk 1]
```diff
@@ -101,7 +101,7 @@ the PyFluent-Visualization documentation.
 
 .. code:: python
 
-   from ansys.fluent.visualization.pyvista import Graphics
+   from ansys.fluent.visualization import Graphics
```
Comment (seanpearsonuk) [inline README.rst:104]: - Import path looks better
- What we're calling `Graphics` here is a proxy / container for renderable solver objects. As such, we might consider a rename in the future.

**Review by mkundu1 — APPROVED** [review]
Approving the refactoring change. We need to improve the animation workflow.

File: examples/00-postprocessing/script_manifold.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+
+cb_init_id = session.events.register_callback("InitializedEvent", initialize_call_back)
+cb_data_read_id = session.events.register_callback(
+    "DataReadEvent", initialize_call_back
+)
+cb_itr_id = session.events.register_callback(
+    "IterationEndedEvent", auto_refersh_call_back_iteration
+)
+
```
[the remark this answers is not in the document]
Comment (prmukherj) [reply examples/00-postprocessing/script_manifold.py:123]: Thank you @mkundu1. Will do it as part of the further enhancement.

**Review by hpohekar — APPROVED** [review]

**Review by ajain-work — APPROVED** [review]

**Review by seanpearsonuk — APPROVED** [review]

**Pull request merged**